linalg/filteredsolver.cpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2010-2025, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "filteredsolver.hpp"
+#include "sparsemat.hpp"
+
+namespace mfem
+{
+
+
+const Operator * FilteredSolver::GetPtAP(const Operator *Aop,
+                                         const Operator *Pop) const
+{
+#ifdef MFEM_USE_MPI
+   const HypreParMatrix * Ah = dynamic_cast<const HypreParMatrix*>(Aop);
+   const HypreParMatrix * Ph = dynamic_cast<const HypreParMatrix*>(Pop);
+   if (Ah && Ph) { return RAP(Ah, Ph); }
+#endif
+#ifdef MFEM_USE_PETSC
+   const PetscParMatrix* Ap = dynamic_cast<const PetscParMatrix*>(Aop);
+   const PetscParMatrix* Pp = dynamic_cast<const PetscParMatrix*>(Pop);
+   if (Ap && Pp) { return RAP(Ap, Pp); }
+#endif
+   const SparseMatrix * Asp = dynamic_cast<const SparseMatrix*>(Aop);
+   const SparseMatrix * Psp = dynamic_cast<const SparseMatrix*>(Pop);
+   if (Asp && Psp)   { return RAP(*Asp, *Psp); }
+
+   return new RAPOperator(*Pop, *Aop, *Pop);
+}
+
+void FilteredSolver::InitVectors() const
+{
+   MFEM_VERIFY(A, "FilteredSolver::MakeSolver: Operator not set");
+   MFEM_VERIFY(P, "FilteredSolver::MakeSolver: Transfer operator not set");
+   MFEM_VERIFY(B, "FilteredSolver::MakeSolver: Solver is not set.");
+   MFEM_VERIFY(S,"FilteredSolver::MakeSolver: Filtered space solver is not set.");
```

Comment:
```suggestion
   MFEM_VERIFY(A, "Operator not set");
   MFEM_VERIFY(P, "Transfer operator not set");
   MFEM_VERIFY(B, "Solver is not set.");
   MFEM_VERIFY(S, "Filtered space solver is not set.");
```

Anyway `MFEM_VERIFY` will give you information about where the error happened